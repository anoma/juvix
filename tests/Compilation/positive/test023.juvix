-- mutual recursion
module test023;

open import Stdlib.Prelude;
open import Stdlib.Data.Nat.Ord;

terminating
f : Nat → Nat;
terminating
g : Nat → Nat;
terminating
h : Nat → Nat;

f x := if (x < 1) 1 (2 * x + g (sub x 1));
g x := if (x < 1) 1 (x + h (sub x 1));
h x := if (x < 1) 1 (x * f (sub x 1));

main : IO;
main :=
  printNatLn (f 5) >>
  printNatLn (f 10) >>
  printNatLn (f 20);

end;
