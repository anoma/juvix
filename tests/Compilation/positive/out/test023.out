32
869
6385109
